### ISSUE TITLE
Add a database connection

### ISSUE THREAD

**Issue by chmanie** [JoinColony/colonyDiscord#2]
Every few blocks, the bot is looking for new events within these blocks. The exact configuration of which events to look for and what to do with them should be stored in a database.

I imagine the following configuration mappings.

There is a per server, per channel setting of the following:

- Colony addresses to watch
- Events to watch

For the fetching of events all the Colony addresses will be retrieved and de-duplicated and stored in an array. The same goes for all the event names, so that when the bot is making the request to the RPC node, all unique Colony addresses and all unique event names are used.

Every time the bot finds an event we need to check if there is a Colony address/event name combination, then fetch the discord server and the channel(s) for that combination to post the event into.

Possible options are:
- A key/value store like redis, with the key being the combination of one colony address and an event (will this mean duplicated values?), or some smarter approach
- A non-relational store like mongodb, using aggregations to uniquely get all colony addresses and events
- A relational database using an appropriate configuration

tl;dr here's the flow:

1. Get all possible events and colony addresses that have been configured from the database
2. Merge them uniquely
3. Send this data to the RPC node when asking for new events
4. For each event that comes back, ask the database for server/channel configurations for the Colony address/event name combination
5. Post the event data into the relevant channel(s)

Goal of this issue would be to pick a data storage approach from above, create a viable data-structure and implement three database CRUD methods:

- One way to store a [`ColonyMultiFilter`](https://github.com/JoinColony/colonySDK/blob/e251099f64cfc79ed4d9a6e04010f11beed50bf6/src/events/ColonyEventManager.ts#L53) in the database (when people add their discord channel and filter configuration)
- One way to get all topics and addresses as flat arrays to send to the RPC node
- One way to get an individual filter by topic and contract address (to get the discord server/channel to post it to)
